### PR TITLE
Pagination feature/protobuf

### PR DESCRIPTION
**Pull request by staheri14**
This PR addresses the following issue https://github.com/status-im/nim-waku/issues/236

### CODE REVIEW

File: tests/v2/test_waku_store.nim [hunk 1]
```diff
@@ -61,26 +61,28 @@ procSuite "Waku Store":
     let 
       index=computeIndex(WakuMessage(payload: @[byte 1], contentTopic:"topic 1"))
       pb=index.encode()
-
-    
-    echo "protobuf ", pb.buffer
-    let decoded_index = Index.init(pb.buffer)
-    echo decoded_index
-    echo index
+      decoded_index = Index.init(pb.buffer)
```
Comment (decanus) [inline tests/v2/test_waku_store.nim:64]: @staheri14 we do not use snake case.

File: tests/v2/test_waku_store.nim [hunk 2]
```diff
@@ -61,26 +61,28 @@ procSuite "Waku Store":
...
     
     check:
       decoded_index.value.receivedTime == index.receivedTime
       decoded_index.value.digest.data == index.digest.data
 
+
+  test "PagingDirection Protobuf encod/init test": 
+    let 
+      pagingDirection=PagingDirection.BACKWARD
```
Comment (decanus) [inline tests/v2/test_waku_store.nim:73]:
```suggestion
      pagingDirection = PagingDirection.BACKWARD
```

File: tests/v2/test_waku_store.nim [hunk 3]
```diff
@@ -61,26 +61,28 @@ procSuite "Waku Store":
     let 
       index=computeIndex(WakuMessage(payload: @[byte 1], contentTopic:"topic 1"))
```
Comment (decanus) [inline tests/v2/test_waku_store.nim:62]:
```suggestion
      index = computeIndex(WakuMessage(payload: @[byte 1], contentTopic:"topic 1"))
```

**Review by decanus — CHANGES_REQUESTED** [review]

**Comment by oskarth** [comment]
@staheri14 can we please rebase this on master and resolve the conflicts?

**Review by oskarth — APPROVED** [review]

File: waku/protocol/v2/waku_store.nim [hunk 4]
```diff
@@ -35,7 +127,11 @@ proc init*(T: type HistoryResponse, buffer: seq[byte]): ProtoResult[T] =
   discard ? pb.getRepeatedField(1, messages)
 
   for buf in messages:
-    msg.messages.add(? WakuMessage.init(buf))
+    msg.messages.add( ? WakuMessage.init(buf))
+
+  var pagingInfoBuffer: seq[byte]
+  discard ? pb.getField(2,pagingInfoBuffer)
+  msg.pagingInfo= ? PagingInfo.init(pagingInfoBuffer)
```
Comment (decanus) [inline waku/protocol/v2/waku_store.nim:134]:
```suggestion
  msg.pagingInfo = ? PagingInfo.init(pagingInfoBuffer)
```

File: waku/protocol/v2/waku_store.nim [hunk 5]
```diff
@@ -16,6 +16,92 @@ logScope:
...
+
+  var data: seq[byte]
+  discard ? pb.getField(1, data)
+
+  # create digest from data
+  index.digest = MDigest[256]()
+  var count = 0
+  for b in data:
+    index.digest.data[count] = b
+    count.inc
```
Comment (decanus) [inline waku/protocol/v2/waku_store.nim:66]: @staheri14 I don't think you need count here?

File: tests/v2/test_waku_store.nim [hunk 6]
```diff
@@ -57,3 +57,106 @@ procSuite "Waku Store":
...
+      # check the correctness of init and encode for an empty HistoryQuery
+      decodedEmptyQuery.isErr == false
+      decodedEmptyQuery.value == emptyQuery
+  
+  test "HistoryResponse Protobuf encod/init test":
+    let
+      wm = WakuMessage(payload: @[byte 1], contentTopic: "topic 1")
+      index = computeIndex(wm)
+      pagingInfo = PagingInfo(pageSize: 1, cursor: index, direction: PagingDirection.BACKWARD)
+      res=HistoryResponse(messages: @[wm], pagingInfo:pagingInfo)
```
Comment (decanus) [inline tests/v2/test_waku_store.nim:144]:
```suggestion
      res = HistoryResponse(messages: @[wm], pagingInfo:pagingInfo)
```

**Review by decanus — CHANGES_REQUESTED** [review]

File: waku/protocol/v2/waku_store.nim (hunk 5 above)
Comment (decanus) [reply waku/protocol/v2/waku_store.nim:66]: you can just do `for count, b in data`

**Pull request closed without merging**